static/app/components/modals/sudoModal.tsx

```diff
@@ -127,95 +155,77 @@ class SudoModal extends Component<Props, State> {
-      retryRequest().then(() => {
-        this.setState({busy: false, showAccessForms: true}, closeModal);
-      });
+    retryRequest().then(() => {
+      setState(prevState => ({...prevState, showAccessForms: true}));
+      closeModal();
     });
   };
 
-  handleError = err => {
-    let errorType = '';
+  const handleError = err => {
+    let newErrorType = ''; // Create a new variable to store the error type
+
     if (err.status === 403) {
       if (err.responseJSON.detail.code === 'no_u2f') {
-        errorType = ErrorCodes.NO_AUTHENTICATOR;
+        newErrorType = ErrorCodes.NO_AUTHENTICATOR;
       } else {
-        errorType = ErrorCodes.INVALID_PASSWORD;
+        newErrorType = ErrorCodes.INVALID_PASSWORD;
       }
     } else if (err.status === 401) {
-      errorType = ErrorCodes.INVALID_SSO_SESSION;
+      newErrorType = ErrorCodes.INVALID_SSO_SESSION;
     } else if (err.status === 400) {
-      errorType = ErrorCodes.INVALID_ACCESS_CATEGORY;
+      newErrorType = ErrorCodes.INVALID_ACCESS_CATEGORY;
     } else if (err === ErrorCodes.NO_AUTHENTICATOR) {
-      errorType = ErrorCodes.NO_AUTHENTICATOR;
+      newErrorType = ErrorCodes.NO_AUTHENTICATOR;
     } else {
-      errorType = ErrorCodes.UNKNOWN_ERROR;
+      newErrorType = ErrorCodes.UNKNOWN_ERROR;
     }
-    this.setState({
-      busy: false,
+
+    setState(prevState => ({
+      ...prevState,
       error: true,
-      errorType,
+      errorType: newErrorType,
       showAccessForms: true,
-    });
+    }));
   };
 
-  handleU2fTap = async (data: Parameters<OnTapProps>[0]) => {
-    this.setState({busy: true});
-
-    const {api, isSuperuser} = this.props;
-
+  const handleU2fTap = async (data: Parameters<OnTapProps>[0]) => {
     try {
       data.isSuperuserModal = isSuperuser;
-      data.superuserAccessCategory = this.state.superuserAccessCategory;
-      data.superuserReason = this.state.superuserReason;
+      data.superuserAccessCategory = state.superuserAccessCategory;
+      data.superuserReason = state.superuserReason;
       await api.requestPromise('/auth/', {method: 'PUT', data});
-      this.handleSuccess();
+      handleSuccess();
     } catch (err) {
-      this.setState({busy: false});
       // u2fInterface relies on this
       throw err;
     }
   };
 
-  getAuthLoginPath(): string {
+  const getAuthLoginPath = (): string => {
     const authLoginPath = `/auth/login/?next=${encodeURIComponent(window.location.href)}`;
     const {superuserUrl} = window.__initialData.links;
     if (window.__initialData?.customerDomain && superuserUrl) {
       return `${trimEnd(superuserUrl, '/')}${authLoginPath}`;
     }
     return authLoginPath;
-  }
+  };
 
-  handleLogout = async () => {
-    const {api} = this.props;
+  const handleLogout = async () => {
     try {
       await logout(api);
     } catch {
       // ignore errors
     }
-    window.location.assign(this.getAuthLoginPath());
+    window.location.assign(getAuthLoginPath());
```

Comment:
Should this be using `router` to redirect the user or do we need to do a full server request to validate auth cookies?